cpu/cpu_linux.go

```diff
@@ -142,23 +141,35 @@ func finishCPUInfo(ctx context.Context, c *InfoStat) {
 		}
 	}
 
-	// override the value of c.Mhz with cpufreq/cpuinfo_max_freq regardless
-	// of the value from /proc/cpuinfo because we want to report the maximum
-	// clock-speed of the CPU for c.Mhz, matching the behaviour of Windows
-	lines, err = common.ReadLines(sysCPUPath(ctx, c.CPU, "cpufreq/cpuinfo_max_freq"))
-	// if we encounter errors below such as there are no cpuinfo_max_freq file,
-	// we just ignore. so let Mhz is 0.
-	if err != nil || len(lines) == 0 {
-		return
-	}
-	value, err = strconv.ParseFloat(lines[0], 64)
-	if err != nil {
-		return
-	}
-	c.Mhz = value / 1000.0 // value is in kHz
-	if c.Mhz > 9999 {
-		c.Mhz = c.Mhz / 1000.0 // value in Hz
+	c.Mhz.Current = fillMhz(ctx, "cur", c)
+	c.Mhz.Min = fillMhz(ctx, "min", c)
+	c.Mhz.Max = fillMhz(ctx, "max", c)
```

Comment:
I would pass


```suggestion
	c.Mhz.Current = fillMhz(ctx, "cpuinfo_cur_freq", c)
	c.Mhz.Min = fillMhz(ctx, "cpuinfo_min_freq", c)
	c.Mhz.Max = fillMhz(ctx, "cpuinfo_max_freq", c)
```

And then I would use

```suggestion
	c.Mhz = Mhz{Current: fillMhz(ctx, "cpuinfo_cur_freq", c), Min: fillMhz(ctx, "cpuinfo_min_freq", c), Max: fillMhz(ctx, "cpuinfo_max_freq", c)}
```